packages/hash/frontend/src/pages/shared/ory-kratos.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  Configuration,
+  V0alpha2Api,
+  SelfServiceError,
+  SelfServiceLoginFlow,
+  SelfServiceRegistrationFlow,
+  SelfServiceBrowserLocationChangeRequiredError,
+  ErrorAuthenticatorAssuranceLevelNotSatisfied,
+  NeedsPrivilegedSessionError,
+} from "@ory/client";
+import { isUiNodeInputAttributes } from "@ory/integrations/ui";
+import { AxiosError } from "axios";
+import { NextRouter, useRouter } from "next/router";
+import {
+  DependencyList,
+  Dispatch,
+  SetStateAction,
+  useEffect,
+  useState,
+} from "react";
+
+type SelfServiceFlow = SelfServiceLoginFlow | SelfServiceRegistrationFlow;
+
+export const oryKratosClient = new V0alpha2Api(
+  new Configuration({
+    /**
+     * Directly connecting to kratos (using "http://127.0.0.1:4433") would prevent the
```

Comment:
I assume that this is the public API that is interfaced with here. Do we allow anyone to access the kratos public API or do we still restrict network access? I ask because it would have some implications on our vercel+AWS setup

Reply:
I think currently there's no restriction on who can make requests to the API. This keeps things simple in our dev configuration so that both the HASH workspace FE and BE can connect to the endpoint.

I think the self-service flows can only work at a specific endpoint however which is specified in the kratos config (so for the local dev environment `localhost:3000`). This is because they rely on kratos performing redirects to various endpoints.